x-pack/legacy/plugins/siem/public/components/ml_popover/ml_popover.tsx

```diff
@@ -0,0 +1,276 @@
+ */
+
+import React, { useContext, useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiDescriptionList,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiIcon,
+  EuiPopover,
+  EuiSpacer,
+  EuiSwitch,
+  EuiText,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import styled from 'styled-components';
+import { setupMlJob, startDatafeeds, stopDatafeeds } from './api';
+import { useJobSummaryData } from './hooks/use_job_summary_data';
+import { useSiemJobs } from './hooks/use_siem_jobs';
+import * as i18n from './translations';
+import { KibanaConfigContext } from '../../lib/adapters/framework/kibana_framework_adapter';
+import { ConfigTemplate, DisplayJob, Job } from './types';
+
+const FilterJobsEuiFlexGroup = styled(EuiFlexGroup)`
+  margin-top: -20px;
+`;
+
+const SpanH4 = styled.h4`
+  display: inline-block;
+`;
+
+const PopoverContentsDiv = styled.div`
+  width: 450px;
+`;
+
+const siemJobPrefix = 'siem-api-';
+
+/**
+ * Config Templates w/ corresponding defaultIndexPattern and jobId's of the SIEM Jobs embedded
+ * in ML. Added as part of: https://github.com/elastic/kibana/pull/39678/files
+ */
+const configTemplates: ConfigTemplate[] = [
+  {
+    name: 'siem_auditbeat_ecs',
+    defaultIndexPattern: 'auditbeat-*',
+    jobs: [
+      `${siemJobPrefix}rare_process_linux_ecs`,
+      `${siemJobPrefix}suspicious_login_activity_ecs`,
+    ],
+  },
+  {
+    name: 'siem_winlogbeat_ecs',
+    defaultIndexPattern: 'winlogbeat-*',
+    jobs: [`${siemJobPrefix}rare_process_windows_ecs`],
+  },
+];
+
+export const getJobsToInstall = (templates: ConfigTemplate[]): string[] =>
+  templates.reduce((jobs: string[], template) => [...jobs, ...template.jobs], []);
+
+export const getConfigTemplatesToInstall = (
+  templates: ConfigTemplate[],
+  installedJobIds: string[],
+  indexPattern: string
+): ConfigTemplate[] =>
+  templates
+    .filter(ct => !ct.jobs.every(ctJobId => installedJobIds.includes(ctJobId)))
+    .filter(ct => indexPattern.indexOf(ct.defaultIndexPattern) >= 0);
+
+export const getJobsToDisplay = (
+  siemJobSummaryData: Job[] | null,
+  embeddedJobIds: string[],
+  showAllJobs: boolean
+): DisplayJob[] =>
+  siemJobSummaryData
+    ? siemJobSummaryData
+        .filter(job => (showAllJobs ? true : embeddedJobIds.includes(job.id)))
+        .map(job => ({
+          title: job.id,
+          description: job.description,
+          isChecked: job.datafeedState === 'started',
+        }))
+    : [];
+
+export const MlPopover = React.memo(() => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [showAllJobs, setShowAllJobs] = useState(false);
+  const [isCreatingJobs, setIsCreatingJobs] = useState(false);
+  const [refetchSummaryData, setRefetchSummaryData] = useState(false);
+  const [, siemJobs] = useSiemJobs(isPopoverOpen ? !refetchSummaryData : refetchSummaryData);
+  const [, siemJobSummaryData] = useJobSummaryData(
```

Comment:
On the next iteration, I think we can simplify things quite a bit by making a composite hook of `useSiemJobs` and `useJobSummaryData`. That should clean up this component quite a bit and remove this `XOR` logic used for updates as well.